src/GraphRequestUtil.ts

```diff
@@ -58,3 +58,99 @@ export const serializeContent = (content: any): any => {
 	}
 	return content;
 };
+
+export class PathUtils {
+	private static hostRe = /([a-z]+.microsoft.[a-z]+)/g;
+	private static versionRe = /(beta|v1.0)/g;
+	private static fullPathRe = /((v1.0|beta).*?(?=\?|$))/g;
+	private static shortPathRe = /(.*?)((?=\?|$))/g;
+	private static queryRe = /\?.*/g;
+
+	private static getPathFromFullPath(url: string): string {
+		const version = PathUtils.getVersionFrom(url);
+		const path = url.match(PathUtils.fullPathRe);
+		if (!path) {
+			return;
+		}
+
+		return path.shift().replace(version, "");
+	}
+
+	private static getPathFromShortPath(url: string) {
```

Comment:
I would propose the following refactor for code reuse.

It also handles strong typing as current functions have empty returns vs string returns. I think empty returns in TS become `undefined` if the return type is `string`, but returning a defaultValue might be better for the consumers of the API. Even if we want to return `undefined`, it is better to be explicit about it.
```
	private static matchRegex(url:string, regex: RegExp, defaultValue: string = "", clearAfterMatch : string = null) : string {
		const matched = url.match(regex);
		if (!matched) {
			return defaultValue;
		}

		if (clearAfterMatch) {
			return matched.shift().replace(clearAfterMatch, "");
		}

		return matched.shift();
	}

	private static getPathFromFullPath(url: string): string {
		const version = this.getVersionFrom(url);
		return this.matchRegex(url, this.fullPathRe, "", version);
	}

	private static getPathFromShortPath(url: string) {
		return this.matchRegex(url, this.shortPathRe);
	}

	public static getHostFrom(url: string): string {
		return this.matchRegex(url, this.hostRe, "graph.microsoft.com");
	}

	public static getVersionFrom(url: string): string {
		return this.matchRegex(url, this.versionRe, "v1.0");
	}

	public static getQueryStringFrom(url: string): string {
		return this.matchRegex(url, this.queryRe, "", "?");
	}
```